packages/browser/src/index.ts

```diff
@@ -1,377 +1,76 @@
-
-    if (!refreshToken) {
-      localStorage.removeItem(refreshTokenKey);
-
-      return;
-    }
-
-    localStorage.setItem(refreshTokenKey, refreshToken);
-  }
-
-  get idToken() {
-    return this._idToken;
-  }
-
-  private set idToken(idToken: Nullable<string>) {
-    this._idToken = idToken;
-
-    const idTokenKey = buildIdTokenKey(this.logtoStorageKey);
-
-    if (!idToken) {
-      localStorage.removeItem(idTokenKey);
+export type {
+  IdTokenClaims,
+  LogtoErrorCode,
+  LogtoRequestErrorBody,
+  LogtoConfig,
+  LogtoClientErrorCode,
+} from '@logto/client';
+export { LogtoError, OidcError, Prompt, LogtoRequestError, LogtoClientError } from '@logto/client';
 
-      return;
-    }
+const requester = async <T>(...args: Parameters<typeof fetch>): Promise<T> => {
+  const response = await fetch(...args);
 
-    localStorage.setItem(idTokenKey, idToken);
+  if (!response.ok) {
+    // Expected request error from server
+    const { code, message } = await response.json<LogtoRequestErrorBody>();
+    throw new LogtoRequestError(code, message);
   }
 
-  // eslint-disable-next-line complexity
-  public async getAccessToken(resource?: string): Promise<string> {
-    if (!this.idToken) {
-      throw new LogtoClientError('not_authenticated');
-    }
-
-    const accessTokenKey = buildAccessTokenKey(resource);
-    const accessToken = this.accessTokenMap.get(accessTokenKey);
-
-    if (accessToken && accessToken.expiresAt > Date.now() / 1000) {
-      return accessToken.token;
-    }
-
-    // Since the access token has expired, delete it from the map.
-    if (accessToken) {
-      this.accessTokenMap.delete(accessTokenKey);
-    }
-
-    /**
-     * Need to fetch a new access token using refresh token.
-     * Reuse the cached promise if exists.
-     */
-    const cachedPromise = this.getAccessTokenPromiseMap.get(accessTokenKey);
+  return response.json<T>();
+};
 
-    if (cachedPromise) {
-      return cachedPromise;
-    }
+const handleRedirect = (url: string) => {
+  window.location.assign(url);
+};
 
-    /**
-     * Create a new promise and cache in map to avoid race condition.
-     * Since we enable "refresh token rotation" by default,
-     * it will be problematic when calling multiple `getAccessToken()` closely.
-     */
-    const promise = this.getAccessTokenByRefreshToken(resource);
-    this.getAccessTokenPromiseMap.set(accessTokenKey, promise);
+const logtoStorageItemKeyPrefix = `logto`;
 
-    const token = await promise;
-    this.getAccessTokenPromiseMap.delete(accessTokenKey);
+class BrowserStorage implements Storage {
+  private readonly storageKey: string;
 
-    return token;
+  constructor(appId: string) {
+    this.storageKey = `${logtoStorageItemKeyPrefix}:${appId}`;
   }
 
-  public getIdTokenClaims(): IdTokenClaims {
-    if (!this.idToken) {
-      throw new LogtoClientError('not_authenticated');
+  getItem(key: 'idToken' | 'refreshToken' | 'accessToken' | 'signInSession'): Nullable<string> {
```

Comment:
nit: The type of the storage key can be exported from `@logto/client`

Reply:
Aha, that is written by coplit, updated

Reply:
Finally, I find myself better than AI. lol